testproject/Assets/Tests/Runtime/MultiprocessRuntime/Helpers/BuildMultiprocessTestPlayer.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.IO;
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEditor.Build.Reporting;
+#endif
+using UnityEngine;
+
+/// <summary>
+/// This is needed as Unity throws "An abnormal situation has occurred: the PlayerLoop internal function has been called recursively. Please contact Customer Support with a sample project so that we can reproduce the problem and troubleshoot it."
+/// when trying to build from Setup() steps in tests.
+/// </summary>
+public class BuildMultiprocessTestPlayer : MonoBehaviour
+{
+    public const string multiprocessBaseMenuName = "MLAPI Multiprocess Test";
+    public const string BuildAndExecuteMenuName = multiprocessBaseMenuName + "/Build - Execute multiprocess tests #%t";
+    public static string buildPath => Path.Combine(Path.GetDirectoryName(Application.dataPath), "Builds/MultiprocessTestBuild");
+    public const string mainSceneName = "MultiprocessTestingScene";
+
+
+#if UNITY_EDITOR
+    [MenuItem(multiprocessBaseMenuName+"/Build Test Player #t")]
+    public static void BuildNoDebug()
+    {
+        var success = Build();
+        if (!success)
+        {
+            throw new Exception("Build failed!");
+        }
+    }
+
+    [MenuItem(multiprocessBaseMenuName+"/Build Test Player in debug mode")]
+    public static void BuildDebug()
+    {
+        var success = Build(true);
+        if (!success)
+        {
+            throw new Exception("Build failed!");
+        }
+    }
+
+    [MenuItem(multiprocessBaseMenuName+"/Delete Test Build")]
+    public static void DeleteBuild()
+    {
+        switch (Application.platform)
+        {
+            case RuntimePlatform.WindowsPlayer:
+            case RuntimePlatform.WindowsEditor:
+                var exePath = $"{buildPath}.exe";
```

Comment:
What about the data folder that lives next to the exe?

Reply:
the goal isn't to cleanup here, it's really to prevent execution of old builds because there was a failure previously.
I would expect workspace cleanup to be done at the yamato/bokken level afterward for all files, not just builds

Reply:
fixed